amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/KinesisDataFetcher.java

```diff
@@ -97,11 +96,29 @@ public KinesisDataFetcher(KinesisAsyncClient kinesisClient, StreamIdentifier str
         this.streamAndShardId = streamIdentifier.serialize() + ":" + shardId;
```

Comment:
Since this constructor is introduced as part of multistreaming story, we can remove this and we can use your new constructor